### ISSUE TITLE
Checking Google Play Games Installation

### ISSUE THREAD

**Issue by ScareCrow95** [playgameservices/play-games-plugin-for-unity#1949]
Hello,

I have just noticed that google login does not work if you do not have an up to date version of google play games installed. I read through the previous posts on this and it appears that you did add a check for this in the code, but still is not functioning as intended.

if I uninstall GPG from my phone and run the app it gets stuck and doesnt move past this :
![screenshot_2017-09-30-23-18-31](https://user-images.githubusercontent.com/18271051/31051797-f2a93730-a63f-11e7-884d-689bd38fa892.png)

if I reinstall and update it it works just fine. Our game is live on playstore and because of this issue several of our players are not able to get in the game.

I tried this code to detect if GPG is upto date or not but it  returns 0 when GPG is disabled. 

` public bool IsPlayServicesAvailable()
        {
            const string GoogleApiAvailability_Classname = 
                "com.google.android.gms.common.GoogleApiAvailability";
            AndroidJavaClass clazz = 
                new AndroidJavaClass(GoogleApiAvailability_Classname);
            AndroidJavaObject obj = 
                clazz.CallStatic<AndroidJavaObject>("getInstance");

            var androidJC = new AndroidJavaClass("com.unity3d.player.UnityPlayer");
            var activity = androidJC.GetStatic<AndroidJavaObject>("currentActivity");

            int value = obj.Call<int>("isGooglePlayServicesAvailable", activity);

            // result codes from https://developers.google.com/android/reference/com/google/android/gms/common/ConnectionResult

            // 0 == success
            // 1 == service_missing
            // 2 == update service required
            // 3 == service disabled
            // 18 == service updating
            // 9 == service invalid
            return value == 0;
        }`

**Comment by Russel1** [comment]
Did you find a solution? @ScareCrow95

**Comment by ScareCrow95** [comment]
Nope, I simply just wait if sign in doesn't happen I show a custom message telling the player to update it's GPG work around but that's the only thing I could think of.

**Comment by yohaan21** [comment]
I am having the same issues .
please let me know if you find any solution

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
